weaviate/collection/collection.py

```diff
@@ -48,6 +53,9 @@ def __init__(
         self.__tenant = tenant
         self.__consistency_level = consistency_level
 
+        self.__iter_object_cache: List[_Object[Properties]] = []
```

Comment:
Would it be a meaningful performance boost if this was a `deque`? Or are `list.pop()` (on L82) and `deque.pop()` basically the same speed?

Reply:
we get a list back from the .fetch_objects() call. I don't think copying it over to use deque brings any improvements.

I also added a test with 2000 objects and it is instant, so I don't think that is an issue